src/plans/components/ProviderSelect.tsx

```diff
@@ -0,0 +1,95 @@
+import { ComponentProps, FC, ReactNode, useMemo } from 'react';
+
+import Select from '@components/common/MtvSelect';
+import { EmptyState } from '@components/EmptyState';
+import { ProviderModelGroupVersionKind, ProviderModelRef, V1beta1Provider } from '@kubev2v/types';
+import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  EmptyStateVariant,
+  Flex,
+  FlexItem,
+  Icon,
+  SelectOption,
+  Title,
+} from '@patternfly/react-core';
+import { isEmpty } from '@utils/helpers';
+import { getResourceUrl } from 'src/modules/Providers/utils/helpers/getResourceUrl';
+import { ExternalLink } from '@components/common/ExternalLink/ExternalLink';
+import { useForkliftTranslation } from '@utils/i18n';
+import { ExclamationCircleIcon } from '@patternfly/react-icons';
+
+type ProviderSelectProps = Pick<ComponentProps<typeof Select>, 'onSelect' | 'status'> & {
+  id: string;
+  value: string;
+  namespace: string | undefined;
+  placeholder?: string;
+  emptyState?: ReactNode;
+};
+
+export const ProviderSelect: FC<ProviderSelectProps> = ({
+  id,
+  value,
+  namespace,
+  placeholder,
+  emptyState,
+  status,
+  onSelect,
+}) => {
+  const { t } = useForkliftTranslation();
+  const [providers, providersLoaded, providerError] = useK8sWatchResource<V1beta1Provider[]>({
+    namespace,
+    groupVersionKind: { ...ProviderModelGroupVersionKind, kind: 'what' },
+    namespaced: true,
+    isList: true,
+  });
+
+  const providersListUrl = useMemo(
+    () =>
+      getResourceUrl({
+        reference: ProviderModelRef,
+        namespace: namespace,
+      }),
+    [namespace],
+  );
+
+  return (
+    <Select
+      id={id}
+      value={value}
+      status={status}
+      onSelect={onSelect}
+      placeholder={!providersLoaded && !providerError ? t('Loading...') : placeholder}
+    >
+      {providerError && providerError instanceof Error ? (
+        <EmptyState variant={EmptyStateVariant.xs}>
+          <Flex direction={{ default: 'column' }} alignItems={{ default: 'alignItemsCenter' }}>
+            <Icon status="danger" size="lg">
+              <ExclamationCircleIcon />
+            </Icon>
+
+            <Title headingLevel="h4">{t('Cannot fetch data')}</Title>
+
+            <FlexItem>{providerError.message}</FlexItem>
+          </Flex>
+        </EmptyState>
+      ) : isEmpty(providers) ? (
+        emptyState || (
+          <EmptyState variant={EmptyStateVariant.xs}>
```

Comment:
we have the new emptystate component u created and can be used here?

Reply:
I am going to remove the common component because its only use is to translate anyways. It doesn't make sense for us to have to remember to use our version every time we use EmptyState, which can have many different variations - and will change even more-so with PFv6.